### PR TITLE
Updated xml docs for Poisson, OLS, and OGD regression trainers.

### PR DESCRIPTION
**Pull request by shmoradims**
Related to #2522

### CODE REVIEW

File: src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs [hunk 1]
```diff
@@ -483,10 +483,10 @@ public IClassificationLoss CreateComponent(IHostEnvironment env)
         }
 
         /// <summary>
-        /// Predict a target using a linear regression model trained with the <see cref="OnlineGradientDescentTrainer"/> trainer.
+        /// Predict a target using a linear regression model trained with the <see cref="OnlineGradientDescentTrainer"/> and advanced options.
```
Comment (sfilipi) [inline src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: >and [](start = 121, length = 3)

specifying

File: src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs [hunk 2]
```diff
@@ -555,7 +555,7 @@ public static LogisticRegressionBinaryTrainer LogisticRegression(this BinaryClas
         /// <param name="l1Regularization">The L1 <a href='tmpurl_regularization'>regularization</a> hyperparameter. Higher values will tend to lead to more sparse model.</param>
         /// <param name="l2Regularization">The L2 weight for <a href='tmpurl_regularization'>regularization</a>.</param>
         /// <param name="optimizationTolerance">Threshold for optimizer convergence.</param>
-        /// <param name="historySize">Memory size for <see cref="Microsoft.ML.Trainers.PoissonRegressionTrainer"/>. Low=faster, less accurate.</param>
+        /// <param name="historySize">Number of previous iterations to remember for estimate of Hessian. Lower values mean faster but less accurate estimates.</param>
```
Comment (sfilipi) [inline src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: >for estima [](start = 80, length = 10)

for the estimate of the Hessian.

File: src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs [hunk 3]
```diff
@@ -573,10 +573,10 @@ public static LogisticRegressionBinaryTrainer LogisticRegression(this BinaryClas
         }
 
         /// <summary>
-        /// Predict a target using a linear regression model trained with the <see cref="Microsoft.ML.Trainers.PoissonRegressionTrainer"/> trainer.
+        /// Predict a target using a linear regression model trained with the <see cref="Microsoft.ML.Trainers.PoissonRegressionTrainer"/> and advanced options.
```
Comment (sfilipi) [inline src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: >and [](start = 139, length = 3)

specifying

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs [hunk 4]
```diff
@@ -48,7 +54,7 @@ public abstract class OptionsBase : TrainerInputBaseWithWeight
             public float OptmizationTolerance = Defaults.OptimizationTolerance;
 
             /// <summary>
-            /// Number of previous iterations to remember for estimate of Hessian.
+            /// Number of previous iterations to remember for estimate of Hessian. Lower values mean faster but less accurate estimates.
```
Comment (sfilipi) [inline src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: > for estimate of Hessian [](start = 57, length = 24)

ditto

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs [hunk 5]
```diff
@@ -71,20 +77,19 @@ public abstract class OptionsBase : TrainerInputBaseWithWeight
             public float StochasticGradientDescentInitilaizationTolerance = 0;
 
             /// <summary>
-            /// Features must occur in at least this many instances to be included
+            /// Determines whether to produce output during training or not.
```
Comment (sfilipi) [inline src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: >Determines whether to produce output during training or not. [](start = 16, length = 60)

should it be more explicit: If set to true produces no output during training. 

Users can imply the correct usage from the parameter name, but why leave it to implying.

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs [hunk 6]
```diff
@@ -71,20 +77,19 @@ public abstract class OptionsBase : TrainerInputBaseWithWeight
...
             /// </summary>
-            /// <remarks>If greater than 1, forces an initialization pass over the data</remarks>
-            //AP removed from now. This requires data transformatio. Perhaps we should handle it as seprate (non-learner) dependant
```
Comment (sfilipi) [inline src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: shall we keep this piece of info? I bet one = true.

File: src/Microsoft.ML.StandardTrainers/Standard/Online/OnlineGradientDescent.cs [hunk 7]
```diff
@@ -23,8 +23,17 @@
 
 namespace Microsoft.ML.Trainers
 {
-
-    /// <include file='doc.xml' path='doc/members/member[@name="OGD"]/*' />
+    /// <summary>
+    /// The <see cref="IEstimator{TTransformer}"/> for training a linear regression model using
+    /// online gradient descent (OGD) for estimating the parameters of the linear regression model.
```
Comment (sfilipi) [inline src/Microsoft.ML.StandardTrainers/Standard/Online/OnlineGradientDescent.cs]: > online gradient descent [](start = 7, length = 24)

feel like those are usually capitalized. 

see here: [https://papers.nips.cc/paper/3319-adaptive-online-gradient-descent.pdf](https://papers.nips.cc/paper/3319-adaptive-online-gradient-descent.pdf)

File: src/Microsoft.ML.StandardTrainers/Standard/Online/OnlineGradientDescent.cs [hunk 8]
```diff
@@ -33,18 +42,24 @@ public sealed class OnlineGradientDescentTrainer : AveragedLinearTrainer<Regress
...
+        /// Options for the <see cref="OnlineGradientDescentTrainer"/>.
+        /// </summary>
         public sealed class Options : AveragedLinearOptions
         {
             [Argument(ArgumentType.Multiple, Name = "LossFunction", HelpText = "Loss Function", ShortName = "loss", SortOrder = 50)]
             [TGUI(Label = "Loss Function")]
             internal ISupportRegressionLossFactory RegressionLossFunctionFactory = new SquaredLossFactory();
 
+            /// <summary>
+            /// A custom <a href="tmpurl_loss">loss</a>.
```
Comment (sfilipi) [inline src/Microsoft.ML.StandardTrainers/Standard/Online/OnlineGradientDescent.cs]: >tmpurl_loss [](start = 34, length = 11)

should probably fix those before V1

File: src/Microsoft.ML.StandardTrainers/Standard/PoissonRegression/PoissonRegression.cs [hunk 9]
```diff
@@ -24,14 +24,26 @@
 
 namespace Microsoft.ML.Trainers
 {
-    /// <include file='doc.xml' path='doc/members/member[@name="PoissonRegression"]/*' />
+    /// <summary>
+    /// The <see cref="IEstimator{TTransformer}"/> for training a Poisson regression model.
+    /// </summary>
+    /// <remarks>
+    /// <a href='https://en.wikipedia.org/wiki/Poisson_regression'>Poisson regression</a> is a parameterized regression method.
+    /// It assumes that the log of the conditional mean of the dependent variable follows a linear function of the dependent variables.
```
Comment (sfilipi) [inline src/Microsoft.ML.StandardTrainers/Standard/PoissonRegression/PoissonRegression.cs]: >linear function [](start = 92, length = 15)

a Poisson distribution and is a linear function of the ...

File: src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs (hunk 1 above)
Comment (shmoradims) [reply src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: i think "and" is good enough and reads well. This is the language pattern I've used for all trainers so far, so I don't want to make recursive changes to completed trainers unless there's an error or major improvement.

---
In reply to: [268037979](https://github.com/dotnet/machinelearning/pull/3059#discussion_r268037979) [](ancestors = 268037979)

File: src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs (hunk 2 above)
Comment (shmoradims) [reply src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: done

---
In reply to: [268038027](https://github.com/dotnet/machinelearning/pull/3059#discussion_r268038027) [](ancestors = 268038027)

File: src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs (hunk 3 above)
Comment (shmoradims) [reply src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: same as before

---
In reply to: [268038068](https://github.com/dotnet/machinelearning/pull/3059#discussion_r268038068) [](ancestors = 268038068)

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs (hunk 4 above)
Comment (shmoradims) [reply src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: done

---
In reply to: [268038103](https://github.com/dotnet/machinelearning/pull/3059#discussion_r268038103) [](ancestors = 268038103)

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs (hunk 6 above)
Comment (shmoradims) [reply src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: diff are not showing well here. The removed text is for CountThreshold which is commented out and I removed it.

---
In reply to: [268038318](https://github.com/dotnet/machinelearning/pull/3059#discussion_r268038318) [](ancestors = 268038318)

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs (hunk 5 above)
Comment (shmoradims) [reply src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: added

---
In reply to: [268038266](https://github.com/dotnet/machinelearning/pull/3059#discussion_r268038266) [](ancestors = 268038266)

File: src/Microsoft.ML.StandardTrainers/Standard/Online/OnlineGradientDescent.cs (hunk 8 above)
Comment (shmoradims) [reply src/Microsoft.ML.StandardTrainers/Standard/Online/OnlineGradientDescent.cs]: in April

---
In reply to: [268038547](https://github.com/dotnet/machinelearning/pull/3059#discussion_r268038547) [](ancestors = 268038547)

File: src/Microsoft.ML.StandardTrainers/Standard/Online/OnlineGradientDescent.cs (hunk 7 above)
Comment (shmoradims) [reply src/Microsoft.ML.StandardTrainers/Standard/Online/OnlineGradientDescent.cs]: You're looking at Title Case in the paper. There are all-lowercase instances in that same paper. But I changed it anyways.

---
In reply to: [268038471](https://github.com/dotnet/machinelearning/pull/3059#discussion_r268038471) [](ancestors = 268038471)

File: src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs [hunk 10]
```diff
@@ -546,7 +546,7 @@ public static LogisticRegressionBinaryTrainer LogisticRegression(this BinaryClas
         }
 
         /// <summary>
-        /// Predict a target using a linear regression model trained with the <see cref="Microsoft.ML.Trainers.PoissonRegressionTrainer"/> trainer.
+        /// Predict a target using a linear regression model trained with the <see cref="Microsoft.ML.Trainers.PoissonRegressionTrainer"/>.
         /// </summary>
```
Comment (rogancarr) [inline src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: ></summary> [](start = 12, length = 10)

Can you please add a remarks section with a discussion of what Poisson regression is? We've had users who are trying to use this for plain linear regression, and this *is* a bit specialized.

Comment (rogancarr) [reply src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: Oh, nevermind -- I see you did this is the base trainer.

---
In reply to: [268295289](https://github.com/dotnet/machinelearning/pull/3059#discussion_r268295289) [](ancestors = 268295289)
